src/unix/socket.h

```diff
@@ -82,11 +82,6 @@ static inline int socket_init(process p, heap h, int domain, int type, u32 flags
         struct sock *s)
 {
     runtime_memset((u8 *) s, 0, sizeof(*s));
-    s->fd = allocate_fd(p, s);
```

Comment:
Nit: I realize s->fd will be zero after the memset, but maybe it's a bit clearer to just return 0 below on success.

Reply:
Agreed, returning s->fd is misleading, I changed it to return 0